source/extensions/access_loggers/common/grpc_access_logger.h

```diff
@@ -289,10 +289,10 @@ class GrpcAccessLoggerCache : public Singleton::Instance,
     if (it != cache.access_loggers_.end()) {
       return it->second;
     }
-    const Grpc::AsyncClientFactoryPtr factory =
-        async_client_manager_.factoryForGrpcService(config.grpc_service(), scope_, false);
     const auto logger = createLogger(
-        config, transport_version, factory->create(),
+        config, transport_version,
+        async_client_manager_.factoryForGrpcService(config.grpc_service(), scope_, false)
+            ->createUncachedRawAsyncClient(),
```

Comment:
Why is gRPC access logger using uncached? I think there is a bunch of cache management code we can tear out from this one and replace with the generic mechanism.

Reply:
Can we introduce the caching mechanism in this PR and allow the code owner to make the changes themselves? That would be less risky.

Reply:
Yes, but can you file bugs for each of the individual places where this needs to be done? Thanks.

Reply:
Also, arguably we are going to have to runtime protect each individual migration here, so worth thinking about how to manage that.

Reply:
Sure, will do it after this PR is merged.